20_1 DataGrid: data source fields can't be 'this' and should not contain prohibited symbols (#1901)
DevExpress/devextreme-documentation: api-reference/_hidden/GridBaseColumn/dataField.md

Binds the column to a field of the [dataSource](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/dataSource.md '{basewidgetpath}/Configuration/#dataSource').

---

> The **columns** array can contain the names of data source fields, which is sufficient if you do not need to specify any other column properties. But if you do, then this array can also contain objects that configure columns. To bind a column object to a data source field, use the **dataField** property. 

Reviewer: This paragraph does not read well, and you should not use `, which`.

> In the example below, the **columns** array contains both names of data source fields and column objects:

Reviewer: ```suggestion
In the example below, the **columns** array contains names of data source fields and column objects:
```

Binds the column to a field of the [dataSource](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/dataSource.md '{basewidgetpath}/Configuration/#dataSource').

---

The **columns** array can contain the names of data source fields, which is sufficient if you do not need to specify any other column properties. But if you do, then this array can also contain objects that configure columns. To bind a column object to a data source field, use the **dataField** property. 

> [note]

Reviewer 2: ```suggestion
[note]

Please review the following notes about data binding:
```

> [note]

Reviewer 2: Move this note to the end.

> - If you create an unbound column with [calculateCellValue](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#calculateCellValue), specify the **columns[]**.[name](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#name) property instead.

Reviewer 2: ```suggestion
- If you create an unbound column (use the [calculateCellValue](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#calculateCellValue) function), specify the **columns[]**.[name](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#name) property instead.
```

    
---

[note]

> Please review the following notes about data binding:

Reviewer: ```suggestion
Review the following notes about data binding:
```

> - If you create an unbound column (use the [calculateCellValue](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#calculateCellValue) function), specify the **columns[]**.[name](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#name) property instead.

Reviewer 2: ```suggestion
- If you create an unbound column (use the [calculateCellValue](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#calculateCellValue) function), specify the **columns[]**.[name](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#name) property instead of **dataField**.
```
